examples/light-switch-app/light-switch-common/light-switch-app.matter

```diff
@@ -1668,10 +1669,41 @@ client cluster ColorControl = 768 {
   readonly attribute int16u remainingTime = 2;
   readonly attribute int16u currentX = 3;
   readonly attribute int16u currentY = 4;
+  readonly attribute enum8 driftCompensation = 5;
+  readonly attribute char_string<254> compensationText = 6;
   readonly attribute int16u colorTemperatureMireds = 7;
   readonly attribute enum8 colorMode = 8;
   attribute bitmap8 options = 15;
   readonly attribute nullable int8u numberOfPrimaries = 16;
+  readonly attribute int16u primary1X = 17;
+  readonly attribute int16u primary1Y = 18;
```

Comment:
This is the client-side cluster version. 
It looks to me like the zap file says that the light app has the client-side cluster enabled for light switch. I am not sure why it does that ... but it does.

We do have an oddity in codegen here: our client side always only uses `src/controller/data_model/controller-clusters.matter` however our codegen client and server side.

I believe ideally we should separate client and server IDLs, however that is likely a different change.

Reply:
Light switches are required by spec to support certain client clusters: On/Off, and depending on the switch Level Control, etc.

> our client side always only uses src/controller/data_model/controller-clusters.matter

_Some_ of our clients use that.  Not all.  Light-switch does not.

Reply:
Again, I think we need to clearly decide what "client side" means in .matter files.

Reply:
Client-side in matter files represents what client side shows up as zap (i.e. side == client).
It was meant to be a representation of what zap contains as configuration.

The issue seems to also be that our usage of "side == client" under zap is unusual ... we seem to care about cluster enabling but not about underlying attributes/commands enabling (is this true?). I am not familiar enough with zap logic to understand this. Logic-wise it seems to me like client-side should be able to use whatever it wants where as server side has granular opinionated control about what it wants to support (i.e. we cannot just codegen everything)

Reply:
> The issue seems to also be that our usage of "side == client" under zap is unusual

Our usage _in some specific clients_.

> We seem to care about cluster enabling but not about underlying attributes/commands enabling

That is the current state of things.  Ideally, we would move to not caring about any of the ZAP state in those clients at all, since they are meant to be "everything" clients and should ideally just generate based on the XML.

I am pretty sure Darwin could make that switch now; I can't speak for the other clients.   We would need to think a bit about whether chip-tool should expose some of the clusters that are currently disabled in the client ZAP config (like the proxy bits)....